### PR TITLE
GEM-40 | add NullOrPattern annotation

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package pl.edu.agh.gem.annotation.nullorpattern
+
+import jakarta.validation.Constraint
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.RUNTIME)
+@MustBeDocumented
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt]: why we add that annotation ?

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt]: image

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt]: should i remove it?

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt]: (NullOrPattern works without this annotation)

Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/annotation/nullorpattern/NullOrPattern.kt]: if we dont use that, and that lib is not public api, IMHO better to remove

**Review by lawos98 — APPROVED** [review]

**Pull request merged**